tests/libwait/wait.go

```diff
@@ -39,29 +37,92 @@ import (
+// These properties can be customized using With* options.
+// If no context is provided, a new one will be created.
+func WaitForVMIPhase(vmi *v1.VirtualMachineInstance, phases []v1.VirtualMachineInstancePhase, opts ...Option) *v1.VirtualMachineInstance {
+	wp := watcher.WarningsPolicy{FailOnWarnings: true}
+	waiting := Waiting{timeout: defaultTimeout, wp: &wp}
+	for _, f := range opts {
+		f(&waiting)
+	}
+
+	if waiting.ctx == nil {
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+		waiting.ctx = ctx
+	}
+
+	return watchVMIForPhase(waiting.ctx, phases, vmi, waiting.timeout, *waiting.wp, waiting.waitForFail)
+}
+
+// WithContext adds a specific context to the waiting struct
+func WithContext(ctx context.Context) Option {
+	return func(waiting *Waiting) {
+		waiting.ctx = ctx
+	}
 }
 
-// Block until the specified VirtualMachineInstance scheduled and return the target node name.
-func WaitForVMIScheduling(obj runtime.Object, seconds int, wp watcher.WarningsPolicy) *v1.VirtualMachineInstance {
-	ctx, cancel := context.WithCancel(context.Background())
-	defer cancel()
-	return waitForVMIPhase(ctx, []v1.VirtualMachineInstancePhase{v1.Scheduling, v1.Scheduled, v1.Running}, obj, seconds, wp, false)
+// WithTimeout adds a specific timeout to the waiting struct
+func WithTimeout(seconds int) Option {
+	return func(waiting *Waiting) {
+		waiting.timeout = seconds
+	}
+}
+
+// WithWarningsPolicy adds a specific warningPolicy to the waiting struct
+func WithWarningsPolicy(wp *watcher.WarningsPolicy) Option {
+	return func(waiting *Waiting) {
+		waiting.wp = wp
+	}
+}
+
+// WithFailOnWarnings sets if the waiting should fail on warning or not
+func WithFailOnWarnings(failOnWarnings bool) Option {
+	return func(waiting *Waiting) {
+		if waiting.wp == nil {
+			waiting.wp = &watcher.WarningsPolicy{}
+		}
+
+		waiting.wp.FailOnWarnings = failOnWarnings
+	}
+}
+
+// WithWarningsIgnoreList sets the warning that will be ignored during the waiting for phase
+// This option will be ignored if the failOnWarnings is false
+func WithWarningsIgnoreList(warningIgnoreList []string) Option {
+	return func(waiting *Waiting) {
+		if waiting.wp == nil {
+			waiting.wp = &watcher.WarningsPolicy{FailOnWarnings: true}
+		}
+
+		waiting.wp.WarningsIgnoreList = warningIgnoreList
+	}
 }
 
-func waitForVMIPhase(ctx context.Context, phases []v1.VirtualMachineInstancePhase, obj runtime.Object, seconds int, wp watcher.WarningsPolicy, waitForFail bool) *v1.VirtualMachineInstance {
-	vmi, ok := obj.(*v1.VirtualMachineInstance)
-	gomega.ExpectWithOffset(1, ok).To(gomega.BeTrue(), "Object is not of type *v1.VMI")
+// WithWaitForFail adds the specific waitForFail to the waiting struct
+func WithWaitForFail(waitForFail bool) Option {
+	return func(waiting *Waiting) {
+		waiting.waitForFail = waitForFail
+	}
+}
 
+// watchVMIForPhase looks at the vmi object, waiting it satisfies the phases with the passed parameters
```

Comment:
nit: it might be worth noting here that it's waited for `v1.Started` implicitly.